src/sources/models.js

```diff
@@ -140,6 +140,12 @@ class Source {
   get authzTagsToPropagate() {
     return new Set();
   }
+  static toString() {
```

Comment:
hah! Did not realise JS had this prototype which you could override. Just like python ;) 

But of course, JS being JS, `console.log(Source)` doesn't work as expected!

Reply:
Yeah, `console.log()` does some "unusual" formatting, despite MDN saying:

> A list of JavaScript objects to output. The string representations of each of these objects are appended together in the order listed and output.

You can force stringification by using the printf-like form of `console.log`:

```
> console.log("%s", Source)
[Source class]
```